jumanji/training/networks/graph_coloring/actor_critic.py

```diff
@@ -0,0 +1,283 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Optional, Sequence
+
+import chex
+import haiku as hk
+import jax.numpy as jnp
+import jax.tree_util as jtree
+
+from jumanji.environments.logic.graph_coloring import GraphColoring, Observation
+from jumanji.training.networks.actor_critic import (
+    ActorCriticNetworks,
+    FeedForwardNetwork,
+)
+from jumanji.training.networks.parametric_distribution import (
+    CategoricalParametricDistribution,
+)
+from jumanji.training.networks.transformer_block import TransformerBlock
+
+
+def make_actor_critic_networks_graph_coloring(
+    graph_coloring: GraphColoring,
+    num_transformer_layers: int,
+    transformer_num_heads: int,
+    transformer_key_size: int,
+    transformer_mlp_units: Sequence[int],
+) -> ActorCriticNetworks:
+    """Make actor-critic networks for the `GraphColoring` environment."""
+    num_actions = graph_coloring.action_spec().num_values
+    parametric_action_distribution = CategoricalParametricDistribution(
+        num_actions=num_actions
+    )
+    policy_network = make_actor_network_graph_coloring(
+        num_actions=num_actions,
+        num_transformer_layers=num_transformer_layers,
+        transformer_num_heads=transformer_num_heads,
+        transformer_key_size=transformer_key_size,
+        transformer_mlp_units=transformer_mlp_units,
+    )
+    value_network = make_critic_network_graph_coloring(
+        num_actions=num_actions,
+        num_transformer_layers=num_transformer_layers,
+        transformer_num_heads=transformer_num_heads,
+        transformer_key_size=transformer_key_size,
+        transformer_mlp_units=transformer_mlp_units,
+    )
+    return ActorCriticNetworks(
+        policy_network=policy_network,
+        value_network=value_network,
+        parametric_action_distribution=parametric_action_distribution,
+    )
+
+
+class GraphColoringTorso(hk.Module):
+    def __init__(
+        self,
+        num_transformer_layers: int,
+        transformer_num_heads: int,
+        transformer_key_size: int,
+        transformer_mlp_units: Sequence[int],
+        name: Optional[str] = None,
+    ):
+        super().__init__(name=name)
+        self.num_transformer_layers = num_transformer_layers
+        self.transformer_num_heads = transformer_num_heads
+        self.transformer_key_size = transformer_key_size
+        self.transformer_mlp_units = transformer_mlp_units
+        self.model_size = transformer_num_heads * transformer_key_size
+
+    def _make_self_attention_mask(self, adj_matrix: chex.Array) -> chex.Array:
+        # Expand on the head dimension.
+        mask = jnp.expand_dims(adj_matrix, axis=-3)
+        return mask
+
+    def embed_nodes(self, nodes: chex.Array) -> chex.Array:
+        nodes_float = nodes.astype(jnp.float32)
+        nodes_leaves = jnp.stack(jtree.tree_leaves(nodes_float), axis=-1)
```

Comment:
Same comment here as below.
```suggestion
        nodes_leaves = nodes_float
```